### PR TITLE
 controller: Fix "changed" logic to account for nodeReady changes

### PR DESCRIPTION
**Pull request by cgwalters**
The `nodeChanged` logic was not accounting for the fact that
the controller takes readiness into account too.  However, rather
than change that function, let's just remove it and extend our
logic below that which was also effectively doing change detection
so it could log it.

This also removes the logging from `nodeReady()` which got very
noisy in status; we now consistently log just changes in the node
controller.

However, the change detection logic was also implicitly ignoring
nodes which didn't appear to be managed by the MCD at all - think
Windows nodes.  Let's explicitly skip nodes that don't have a `currentConfig`
annotation.

### CODE REVIEW

**Comment by cgwalters** [comment]
OK this one is rebased :surfing_man: on master which has #719 now.

**Comment by cgwalters** [comment]
Ok this is rebased :surfing_man: on master which as #719 now.

File: pkg/controller/node/status.go [hunk 1]
```diff
@@ -153,32 +171,32 @@ func getReadyMachines(currentConfig string, nodes []*corev1.Node) []*corev1.Node
...
 	for i := range node.Status.Conditions {
 		cond := &node.Status.Conditions[i]
 		// We consider the node for scheduling only when its:
 		// - NodeReady condition status is ConditionTrue,
 		// - NodeOutOfDisk condition status is ConditionFalse,
 		// - NodeNetworkUnavailable condition status is ConditionFalse.
 		if cond.Type == corev1.NodeReady && cond.Status != corev1.ConditionTrue {
-			glog.Infof("Node %s is reporting NotReady", node.Name)
-			return false
+			return fmt.Errorf("Node %s is reporting NotReady", node.Name)
```
Comment (kikisdeliveryservice) [inline pkg/controller/node/status.go]: nit: these should be lowercase error messages. we can fix later.

File: pkg/controller/node/status.go [hunk 2]
```diff
@@ -153,32 +171,32 @@ func getReadyMachines(currentConfig string, nodes []*corev1.Node) []*corev1.Node
...
 	if node.Spec.Unschedulable {
-		glog.Infof("Node %s is reporting Unschedulable", node.Name)
-		return false
+		return fmt.Errorf("Node %s is reporting Unschedulable", node.Name)
 	}
-	return true
+	return nil
+}
+
+func isNodeReady(node *corev1.Node) bool {
```
Comment (kikisdeliveryservice) [inline pkg/controller/node/status.go]: couldn't we just return bool, error in checkNodeReady()?

File: pkg/controller/node/node_controller.go [hunk 3]
```diff
@@ -188,31 +188,63 @@ func (ctrl *Controller) updateNode(old, cur interface{}) {
...
 	}
 	if pool == nil {
 		return
 	}
 	glog.V(4).Infof("Node %s updated", curNode.Name)
 
-	oldReady := isNodeReady(oldNode)
-	newReady := isNodeReady(curNode)
+	var changed bool
+	oldReadyErr := checkNodeReady(oldNode)
```
Comment (kikisdeliveryservice) [inline pkg/controller/node/node_controller.go]: later: we could make this a small function instead of repeating code

**Comment by kikisdeliveryservice** [comment]
went through this and LGTM

/lgtm

**Comment by runcom** [comment]
/hold

 for beta I guess

File: pkg/controller/node/status.go (hunk 2 above)
Comment (cgwalters) [reply pkg/controller/node/status.go]: Yeah, though keeping the API the same involved less code churn for the callers.

File: pkg/controller/node/status.go [hunk 4]
```diff
@@ -75,8 +77,12 @@ func calculateStatus(pool *mcfgv1.MachineConfigPool, nodes []*corev1.Node) mcfgv
 		readyMachineCount == machineCount &&
 		unavailableMachineCount == 0 {
 		//TODO: update api to only have one condition regarding status of update.
-		supdated := mcfgv1.NewMachineConfigPoolCondition(mcfgv1.MachineConfigPoolUpdated, corev1.ConditionTrue, fmt.Sprintf("All nodes are updated with %s", pool.Status.Configuration.Name), "")
+		updatedMsg := fmt.Sprintf("All nodes are updated with %s", pool.Status.Configuration.Name)
+		supdated := mcfgv1.NewMachineConfigPoolCondition(mcfgv1.MachineConfigPoolUpdated, corev1.ConditionTrue, updatedMsg, "")
```
[the remark this answers is not in the document]
Comment (cgwalters) [reply pkg/controller/node/status.go]: I think so, I didn't introduce the term though.

**Comment by cgwalters** [comment]
This was merged as part of https://github.com/openshift/machine-config-operator/pull/701

**Pull request closed without merging**